Bone breaking rework

DrCrawler opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Fracturing a body part and the likelihood of it now depends on the actual force of the attack, becoming easier and more likely to fracture the more your body part is damaged.
This does mean that you can actually break a bone from full health, but right now that requires receiving 50+ brute damage in a single attack, which is exceedingly rare.
This also means that you can no longer break bones from something like decompression, hail, sand storms etc., since it deals less than 5 damage per individual proc call.

Force check for being able to fracture:
- Damage taken >= (Damage Threshold - clamp((Existing damage * 0.5), 0, Damage Threshold - 5)
- Where Damage Threshold = 40 (45 for head, 25 for everything else)

Formula for percentage chance of fracture:
- Raw Force of Weapon/Projectile + Existing Damage * 0.5

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Fractures have felt a little unintuitive, mechanically shallow and nonsensical. This aims to solve that through tying the ability to fracture and the probability of inflicting a fracture to the received damage, existing damage and force of the attack.
This also gives armor more importance/usefulness, since it decreases damage taken, which increases the raw attack force required to create a fracture.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
balance: Fractures now rely more on received damage and the force of the attack.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DrCrawler committed:
mmm, milm

DrCrawler committed:
woops, guidelines

DrCrawler committed:
adjustment

DrCrawler committed:
forgot to change comment

DrCrawler committed:
Forgot that this exists

DrCrawler commented:
~~Hold on, forgot skulls.~~
edit: solved.

Apogee-dev commented:
Thank god.

rye-rice commented:
🙏

Rug-Dealer commented:
no more pranking shipmates by breaking their ribs with thrown items

DrCrawler committed:
BOOM

Thank you @Rug-Dealer

thgvr commented:
I'd like to see splints usable on chest with this as well

DrCrawler commented:
> I'd like to see splints usable on chest with this as well

Will try.

SomeguyManperson commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -91,8 +91,6 @@
 	if(splint_fracture) //Check if it's a splint and the bone is broken
-		if(affecting.body_part in list(CHEST, HEAD)) // Check if it isn't the head or chest
-			to_chat(user, span_warning("You can't splint that bodypart!"))
 		else if(affecting.bone_status == BONE_FLAG_SPLINTED) // Check if it isn't already splinted
```
```suggestion
		if(affecting.bone_status == BONE_FLAG_SPLINTED) // Check if it isn't already splinted
```

DrCrawler replied:
woops

DrCrawler committed:
woops!

Co-authored-by: Theos <theubernyan@gmail.com>
Signed-off-by: DrCrawler <62419802+DrCrawler@users.noreply.github.com>

DrCrawler committed:
Merge branch 'master' into calcium

DrCrawler committed:
Merge branch 'master' into calcium

DrCrawler commented:
I should make it easier to break limbs.

Overteh commented:
Something is definitely wrong, elzuose was beaten with a sledgehammer almost to death, but she didn't get a single fracture

DrCrawler committed:
calcium deficiency

- Separate bone breaking thresholds for chest and arms or legs.
- Breaking a rib noticeably easier, although only a limited number of ammo is capable of breaking ribs in 1 shot at the moment. (Counterable with armor)
- Removes break_modifier from the receive_damage and apply_damage procs, redundant with new bone breaking math. Replaced with attack_force var in-proc.

DrCrawler committed:
Merge branch 'calcium' of https://github.com/DrCrawler/Shiptest into calcium

DrCrawler commented:
Will probably need another TM while I tune the math.

DrCrawler committed:
Merge branch 'master' into calcium

generalthrax commented:
When simple mobs die, it seems the body that drops has all bones on it break instantly. Unsure if intended or not

DrCrawler committed:
One more glass of milk

Makes limbs slightly more durable

DrCrawler commented:
> When simple mobs die, it seems the body that drops has all bones on it break instantly. Unsure if intended or not

I've only seen that happen with legions, since they do adjustbruteloss(1000) on infection. So far I've been told to leave it, so I'm gonna wait and see if this is a glaring issue or not.

DrCrawler committed:
Merge branch 'master' into calcium

DrCrawler commented:
I **think** the PR is nearing a merge-ready state? So far feedback, as little as there is, has been mostly positive and there's no bugs associated with this, to my knowledge. Unless something earth-shattering gets uncovered during saturday - my work is finished.
Keep it open until the end of april 13th (sunday), though, just in case something comes up.
